cookbook-react/src/recipes/hooks/useUniqueSelection/index.ts

```diff
@@ -0,0 +1,26 @@
+import { useState } from 'react';
+
+export type OptionId = string | number;
+
+interface Params<Id> {
+  onChange: (id: Id) => void;
+  active?: Id;
+  initialValue?: Id;
+}
+
+const useUniqueSelection = <Id extends OptionId>({ onChange, active, initialValue }: Params<Id>) => {
```

Comment:
In general, do you need a generic for Id? Or could you alway use `OptionId` or `string | number`

Reply:
It's not the same. If I use OptionId or `string | number` then the `handleChange` callback returned has a parameter of type `string | number`, while if I use a generic value extending from OptionId, the value of this callback is of the specific type you are effectevely going to use (string or number).